client/helpers.go

```diff
@@ -216,14 +245,14 @@ func addKeyForRole(kdb *keys.KeyDB, role string, key data.PublicKey) error {
 // signs and serializes the metadata for a canonical role in a tuf repo to JSON
 func serializeCanonicalRole(tufRepo *tuf.Repo, role string) (out []byte, err error) {
 	var s *data.Signed
-	switch role {
-	case data.CanonicalRootRole:
+	if role == data.CanonicalRootRole {
```

Comment:
Ah yeah, you can't do the assignment. You could switch that one to `tufRepo.Targets[role] != nil` as an explicitly set nil value would be just as invalid as an unset value for us.